### DISCUSSION TITLE
Support for displaying Callbacks documentation

### DISCUSSION THREAD

**Discussion by sidtheblue** [Azure/api-management-developer-portal#2539] [Feature request]
The operations page does not display details of "Callbacks" section included in Open API 3.0:
https://swagger.io/docs/specification/callbacks/

Please could any callbacks defined against an operation be displayed in the new developer portal, in a similar way to Swagger UI

**Comment by mikebudzynski** [comment]
@sidtheblue Could you please let us know if you're able to see callbacks in the OpenAPI file generated by API Management, after you create an API by importing your own OpenAPI file with callbacks defined?

You can access the API Management generated file through Azure portal (https://docs.microsoft.com/en-us/azure/api-management/edit-api#update-the-swagger) or through the developer portal (download dropdown).

**Comment by sidtheblue** [comment]
Hi, the answer is no. APIM seemingly "ignores" the callbacks block in Open API 3 specs. When you then choose to export the API Definition in the classic (or new) developer portal, the spec that is returned omits the callbacks section that was originally included in the imported swagger.

**Comment by mikebudzynski** [comment]
@sidtheblue - Thanks for checking. We first need to implement support for `callback` in API Management (I took a note of that) and then include it on the developer portal.

Our goal for GA of the new portal is feature parity with the old one. We'll look again into this request at a later date.

**Comment by mikebudzynski** [comment]
@sidtheblue We're moving all the feature requests to our Feedback Forum. Could you please open this feature request again at https://aka.ms/apimwish?

**Comment by apaulro** [comment]
Any updates on this?

**Comment by mikebudzynski** [comment]
Reopening this issue, since we're now tracking feature requests on GitHub. There are no updates.

**Comment by swjensen** [comment]
It would be great, if this feature request could be prioritized. It doesn't work well with our partners that we now have to send them separate Open Api documents to document the callbacks in our solutions.

**Comment by janoosthoek** [comment]
> It would be great, if this feature request could be prioritized. It doesn't work well with our partners that we now have to send them separate Open Api documents to document the callbacks in our solutions.

We are doing the same thing, was hoping this would be included now we have support to import OpenAPI version 3.1

**Comment by Tapanila** [comment]
Any news on this?

**Comment by DClark-Landmark** [comment]
Any news on this, we now have to migrate to the new developer portal and Callbacks is still not supported.

**Comment by timotanner** [comment]
I'd also like to hear the timeline for the callbacks feature as it's been part of OpenAPI Specification since version 3.0. It would make sense to preserve the callbacks documentation in the portal hosted API description, instead of dropping it.

**Comment by aczarkowski** [comment]
Any update on this? We have exactly the same problem where we need to send a separate openapi spec file to clients, which defies the purpose of the developer portal

**Comment by timotanner** [comment]
Any news or timeline regarding this feature?

**Comment by timotanner** [comment]
Any light at the end of the tunnel for this one? Our requirement is only documentation of callbacks to API clients, without need to implement any actual callback functionalities in the API gateway.

**Comment by crsbrns** [comment]
Has there been any progress on this? It's a feature we need for the project I'm currently working on. Would be great if we didn't have to use another docs provider.
